rest/model/conversions.go

```diff
@@ -40,40 +40,48 @@ type convertFnOpts struct {
 // to find the correct generated type conversion function. It returns the name of the conversion function
 // and the name of the inverse conversion function
 func conversionFn(opts convertFnOpts, generatedConversions map[typeInfo]string) (convertFuncs, error) {
+	info := typeInfo{
+		OutType:  opts.outType,
+		OutIsPtr: opts.outIsPtr,
+	}
 	if _, inIsPrimitive := opts.in.(*types.Basic); inIsPrimitive {
-		return simplePtrs(opts.in.String(), opts.outType, false, opts.outIsPtr, generatedConversions)
+		info.InType = opts.in.String()
+		return simplePtrs(info, generatedConversions)
 	}
 	if intype, inIsPtr := opts.in.(*types.Pointer); inIsPtr {
 		value, isPrimitive := intype.Elem().(*types.Basic)
 		if isPrimitive {
-			return simplePtrs(value.String(), opts.outType, true, opts.outIsPtr, generatedConversions)
+			info.InType = value.String()
+			info.InIsPtr = true
+			return simplePtrs(info, generatedConversions)
 		}
 		underlyingStruct, isStruct := intype.Elem().(*types.Named)
 		if isStruct {
 			return objectTypeFnName(underlyingStruct, opts.outIsPtr)
 		}
 		return convertFuncs{}, errors.New("complex pointers not implemented yet")
 	}
+	info.InType = opts.in.String()
```

Comment:
I think you can just set this at the top, since the only case where it's not true is line 48 which sets it manually. (And then line 54 can be removed)